### PR TITLE
Improve hooks page ui a bit

### PR DESCRIPTION
**Pull request by imbstack**
This does 3 things:

1. Updates the text on the "email if failed" section to help indicate that this is not for failed tasks.
2. Make it so that triggering the hook adds a new link to the new task.
3. Updates react-treeview because chenglou/react-treeview#35 was annoying me

Of the three, (2) is the one I am the least sure about. It seems like I want the UI to react that way, but perhaps the way I've gone about it is un-react-y or something.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,6 @@
     "font-awesome-svg-png": "^1.1.5",
     "highlight.js": "^9.6.0",
     "hterm-umd": "^1.0.1",
-    "js-yaml": "^3.7.0",
```
Comment (eliperelman) [inline package.json]: ._.

File: src/hooks/hookeditor.jsx [hunk 2]
```diff
@@ -57,37 +57,22 @@ const HookStatusDisplay = React.createClass({
...
+    hookStatus: React.PropTypes.object,
+    refreshHookStatus: React.PropTypes.func.isRequired,
   },
 
   render() {
-    const waitFor = this.renderWaitFor('hookStatus');
-
-    if (waitFor) {
-      return waitFor;
+    const stat = this.props.hookStatus;
```
Comment (eliperelman) [inline src/hooks/hookeditor.jsx]: Insert line after declaration.

File: src/hooks/hookeditor.jsx [hunk 3]
```diff
@@ -645,12 +637,24 @@ const HookEditView = React.createClass({
...
       .triggerHook(this.props.currentHookGroupId, this.props.currentHookId, {})
+      .then(this.refreshHookStatus)
       .catch(error => this.setState({error}));
   },
 
+  refreshHookStatus() {
+    this.setState({hookStatus: null});
+    return this.hooks
+      .getHookStatus(this.props.currentHookGroupId, this.props.currentHookId)
+    .then(stat => {
```
Comment (eliperelman) [inline src/hooks/hookeditor.jsx]: Indent this group 2 more spaces.

File: package.json (hunk 1 above)
Comment (imbstack) [inline package.json]: So it turns out this was in `package.json` twice? I think when I `yarn add`'ed the new version of treeview, it removed the duplicate. It's down on line 33 below.

Comment (eliperelman) [inline package.json]: Yeah, just funny that this was here and we didn't notice.

**Pull request merged**